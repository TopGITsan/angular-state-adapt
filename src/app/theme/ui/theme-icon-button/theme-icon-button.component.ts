import { nextThemeChange$ } from '@theme/theme.actions';
import { AsyncPipe } from '@angular/common';
import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { provideTranslocoScope, TranslocoDirective } from '@jsverse/transloco';
import { ThemeStoreService } from '@theme/theme-store/theme-store.service';
import { inlineLoader } from '@shared/i18n/inline-loader.function';
import { ThemeIconMap } from '../../theme.type';

@Component({
  selector: 'app-theme-icon-button',
  imports: [MatButtonModule, MatIconModule, AsyncPipe, TranslocoDirective],
  providers: [
    ThemeStoreService,
    provideTranslocoScope({
      scope: 'theme',
      loader: inlineLoader(
        (lang: string, folder: string) =>
          import(`../../${folder}/${lang}.json`),
      ),
    }),
  ],
  template: `
    @let theme = theme$ | async;
    <ng-container *transloco="let t">
      <button
        class="mx-2"
        mat-icon-button
        attr.aria-label="{{ t('theme.ariaLabel', { theme }) }}"
        title="{{ t('theme.title') }}"
        (click)="changeToNextTheme$.next()"
      >
        <mat-icon>{{ Theme[theme ?? ''] }}</mat-icon>
      </button>
    </ng-container>
  `,
  styles: ``,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ThemeIconButtonComponent {
  readonly Theme = ThemeIconMap;

  readonly themeStore = inject(ThemeStoreService);
  readonly theme$ = this.themeStore.store.theme$
  readonly changeToNextTheme$ = nextThemeChange$;
}
