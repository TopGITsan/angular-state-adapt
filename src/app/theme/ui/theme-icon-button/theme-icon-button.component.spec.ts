import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ThemeIconButtonComponent } from './theme-icon-button.component';
import { defaultStoreProvider } from '@state-adapt/angular';
import { provideExperimentalZonelessChangeDetection } from '@angular/core';
import { getTranslocoTestingModule } from '@internalization/get-transloco-testing-module.function';

describe('ThemeIconButtonComponent', () => {
  let component: ThemeIconButtonComponent;
  let fixture: ComponentFixture<ThemeIconButtonComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ThemeIconButtonComponent, getTranslocoTestingModule()],
      providers: [
        defaultStoreProvider,
        provideExperimentalZonelessChangeDetection(),
      ],
    }).compileComponents();

    fixture = TestBed.createComponent(ThemeIconButtonComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });
});
