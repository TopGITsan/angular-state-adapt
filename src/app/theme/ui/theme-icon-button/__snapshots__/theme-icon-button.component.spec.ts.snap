// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeIconButtonComponent should create 1`] = `
<app-theme-icon-button
  Theme={[Function Object]}
  changeToNextTheme$={[Function Source]}
  theme$={[Function Observable]}
  themeStore={[Function ThemeStoreService]}
>
  <button
    class="mx-2 mdc-icon-button mat-mdc-icon-button"
    mat-icon-button=""
    mat-ripple-loader-centered=""
    mat-ripple-loader-class-name="mat-mdc-button-ripple"
    mat-ripple-loader-uninitialized=""
  >
    <span
      class="mat-mdc-button-persistent-ripple mdc-icon-button__ripple"
    />
    <mat-icon
      aria-hidden="true"
      class="mat-icon notranslate"
      role="img"
    >
    </mat-icon>
    <span
      class="mat-focus-indicator"
    />
    <span
      class="mat-mdc-button-touch-target"
    />
  </button>
</app-theme-icon-button>
`;
